bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDClientConfigurationTest.java

```diff
@@ -1,40 +1,127 @@
 package gov.cms.ab2d.bfd.client;
 
+import ca.uhn.fhir.rest.client.exceptions.FhirClientConnectionException;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.configuration.ConfigurationProperties;
+import org.mockserver.integration.ClientAndServer;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
+import javax.net.ssl.SSLHandshakeException;
+import java.io.File;
 import java.io.IOException;
 import java.net.URISyntaxException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.nio.file.StandardCopyOption;
-import java.security.KeyStore;
+import java.net.URL;
+import java.util.List;
 
-import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static gov.cms.ab2d.bfd.client.MockUtils.getRawJson;
+import static gov.cms.ab2d.fhir.FhirVersion.STU3;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.context.support.TestPropertySourceUtils.addInlinedPropertiesToEnvironment;
 
 @ExtendWith(SpringExtension.class)
-@SpringBootTest(classes = SpringBootApp.class, properties = "bfd.keystore.location=/tmp/bb.keystore")
+@SpringBootTest(classes = SpringBootApp.class)
+@ContextConfiguration(initializers = {BFDClientConfigurationTest.PropertyOverrider.class})
 public class BFDClientConfigurationTest {
 
     @Autowired
-    private KeyStore keyStore;
+    private BFDClientImpl bbc;
```

Comment:
Either change to an interface or add a comment describing why the Impl is necessary.